File: AtCoder/abc/abc200/c.cpp
#include <bits/stdc++.h>
#define rep(i, n) for(int i = 0; i < (n); ++i)
#define rep1(i, n) for(int i = 1; i < (n); ++i)
using namespace std;
using ll = long long;

int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    int n;
    cin >> n;
    vector<ll> a(n);
    map<ll, ll> mp;
    ll mxa = 0;
    vector<vector<ll>> cnt(200);
    rep(i, n){
        cin >> a[i];
        ++mp[a[i]];
        mxa = max(mxa, a[i]);
        cnt[]
    }

    ll res = 0;
    for(auto k: mp) res += k.second * (k.second - 1) / 2;

    for(auto k: mp){
        ll from = k.first;
        for(ll i = 1; i < mxa / 200 + 1; ++i){
            ll to = k.first + 200 * i;
            cout << from << " " << to << endl;
            res += mp[from] * mp[to];
        }
    }

    cout << res << endl;

    return 0;
}

File: AtCoder/abc/abc200/c.py
n = int(input())
a = list(map(int, input().split()))
mxa = max(a)

mp = dict()
for e in a:
    if e in mp: mp[e] += 1
    else: mp[e] = 1

cnt = [[] for _ in range(200)]
for k in mp:
    cnt[k%200].append(mp[k])

res = 0
for k in mp:
    res += mp[k] * (mp[k]-1) // 2

for st in range(200):
    b = cnt[st]
    
    m = len(b)
    if m < 2: continue
    # print(st, b)
    now = b[-1]
    for i in range(m-2, -1, -1):
        res += b[i] * now
        now += b[i]

print(res)